Source/ACE/Entity/Lifestone.cs

```diff
@@ -1,63 +1,88 @@
+                Use = "Use this item to set your resurrection point.";
+
+            var useString = new AceObjectPropertiesString();
+            useString.AceObjectId = Guid.Full;
+            useString.PropertyId = (ushort)PropertyString.Use;
+            useString.PropertyValue = Use;
+            lifestonePropertiesString.Add(useString);
         }
 
+        private List<AceObjectPropertiesString> lifestonePropertiesString = new List<AceObjectPropertiesString>();
+        
         public override void HandleActionOnUse(ObjectGuid playerId)
         {
-            // All data on a lifestone is constant -- therefore we just run in context of player
             ActionChain chain = new ActionChain();
             CurrentLandblock.ChainOnObject(chain, playerId, (WorldObject wo) =>
             {
                 Player player = wo as Player;
-                string serverMessage = null;
-                // validate within use range, taking into account the radius of the model itself, as well
-                SetupModel csetup = SetupModel.ReadFromDat(SetupTableId.Value);
-                float radiusSquared = (UseRadius.Value + csetup.Radius) * (UseRadius.Value + csetup.Radius);
-
-                // Run this animation...
-                // Player Enqueue:
-                var motionSanctuary = new UniversalMotion(MotionStance.Standing, new MotionItem(MotionCommand.Sanctuary));
-
-                var animationEvent = new GameMessageUpdateMotion(player.Guid,
-                                                                 player.Sequences.GetCurrentSequence(Network.Sequence.SequenceType.ObjectInstance),
-                                                                 player.Sequences, motionSanctuary);
-
-                // This event was present for a pcap in the training dungeon.. Why? The sound comes with animationEvent...
-                var soundEvent = new GameMessageSound(Guid, Sound.LifestoneOn, 1);
-
-                if (player.Location.SquaredDistanceTo(Location) >= radiusSquared)
+                if (player == null)
                 {
-                    serverMessage = "You wandered too far to attune with the Lifestone!";
+                    return;
                 }
+
+                if (!player.IsWithinUseRadiusOf(this))
+                    player.DoMoveTo(this);
                 else
                 {
-                    player.SetCharacterPosition(PositionType.Sanctuary, player.Location);
+                    ActionChain useObjectChain = new ActionChain();
 
-                    // create the outbound server message
-                    serverMessage = "You have attuned your spirit to this Lifestone. You will resurrect here after you die.";
-                    player.HandleActionMotion(motionSanctuary);
-                    player.Session.Network.EnqueueSend(soundEvent);
-                }
+                    useObjectChain.AddAction(player, () =>
+                    {
+                        Activate(player);
```

Comment:
Your right that the context should be run on the thing it is modifying.  When you create this action chain to call Activate, your already in the context of the Player, and there is no delay in the action chain before calling activate.

Its just a redundant action chaining.  Its not "wrong", but its confusing.  Your explicitly delaying work, but with no reason.  The act of delaying work indicates there is a reason.